File: firebase/functions/src/repositories/feeling.ts
import { feelingsRef } from '../firestore-refs/firestoreRefs'
import { Feeling } from '../models/feeling'

export class FeelingRepository {
    /** 指定した votingEvent の feeling を全件取得する */
    async fetchFeelings({ roomId, votingEventId }: { roomId: string; votingEventId: string }): Promise<Feeling[]> {
        const qs = await feelingsRef({ roomId, votingEventId }).get()
        return qs.docs.map((a) => a.data())
    }
}

File: lib/features/voting_event/feeling.dart
import 'package:hooks_riverpod/hooks_riverpod.dart';
import 'package:tuple/tuple.dart';

import '../../models/feeling.dart';
import '../../repositories/firestore/feeling_repository.dart';

final myFeelingsProvider = FutureProvider.autoDispose
    .family<List<Feeling>, Tuple3<String, String, String>>((ref, param) async {
  final feelings =
      await ref.watch(feelingRepositoryProvider).getFeelingsByUserId(
            roomId: param.item1,
            votingEventId: param.item2,
            userId: param.item3,
          );
  return feelings;
});
